src/SegmentServiceProvider.php

```diff
@@ -35,7 +35,12 @@ public function boot()
         $this->setupConfig();
 
         if ($writeKey = $this->app->config->get('segment.write_key')) {
-            Segment::init($writeKey);
+            $options = [
+                'consumer' => $this->app->config->get('segment.consumer') ?? null,
```

Comment:
No need for the `?? null` I think.

Reply:
What if you were to just do an array_only on `$this->app->config->get('segment')`, and then a filter. That'd give you the correct options?